helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -358,8 +365,9 @@ public void scheduleRebalance(long rebalanceTime) {
   /**
    * Schedule an on demand rebalance pipeline.
    * @param delay
+   * @param shouldRefreshCache true if refresh the cache before scheduling a rebalance.
    */
-  public void scheduleOnDemandRebalance(long delay) {
+  public void scheduleOnDemandRebalance(long delay, boolean shouldRefreshCache) {
```

Comment:
Constructor change in a public method may cause backward-compatibility issues. Could we keep the original public method and add this overloaded one?

If so, what should the default behavior be for `public void scheduleOnDemandRebalance(long delay)`?

Reply:
Good catch. The old behavior is shouldRefreshCache = true.